test-data/unit/check-overloading.test

```diff
@@ -5001,3 +5001,18 @@ def f(x):
 
 reveal_type(f(g([])))  # E: Revealed type is 'builtins.list[builtins.int]'
 [builtins fixtures/list.pyi]
+
+[case testTypeCheckOverloadCoroutine]
+from types import coroutine
+from typing import overload
+@overload
+@coroutine
+def f(x: int) -> None: ...
+@overload
+@coroutine
+def f(x: str) -> None: ...
+def f(x): pass
```

Comment:
I wonder if we should maybe also include a check that enforces that `@coroutine` is included in the implementation as well? Or more generally, mandate that decorators used consistently across all overload signatures + the implementation.

Normally, we don't check whether the implementation signature actually matches the given overload variants because it's too difficult to do so. But it might be worth the extra effort to make sure at least decorators in general are applied consistently: it's not as hard to check, and not including the `@coroutine` decorator on the implementation can result in some potentially confusing errors at runtime.

The fact that we currently don't do this for `@abstractmethod` is an oversight, I think -- it doesn't really make a major difference at runtime whether `f` *actually* is abstract or not (especially since mypy's abstract checks largely supersede the same ones done at runtime by the abc module), so probably nobody bothered reporting this.

This is probably out-of-scope for this PR though. Up to you if you want to try tackling this or not -- but if you don't have time, could you file an issue so we can keep track of this?